chia/util/condition_tools.py

```diff
@@ -25,12 +25,28 @@ def parse_sexp_to_condition(
     Takes a ChiaLisp sexp and returns a ConditionWithArgs.
     If it fails, returns an Error
     """
-    as_atoms = sexp.as_atom_list()
-    if len(as_atoms) < 1:
+    i = sexp.as_iter()
+    opcode = next(i)
+    if opcode is None:
         return Err.INVALID_CONDITION, None
-    opcode = as_atoms[0]
-    opcode = ConditionOpcode(opcode)
-    return None, ConditionWithArgs(opcode, as_atoms[1:])
+    op = opcode.atom
+    if op is None or len(op) != 1:
+        return Err.INVALID_CONDITION, None
+
+    # since the ConditionWithArgs only has atoms as the args, we can't parse
+    # hints and memos with this function
+    vars: List[bytes] = []
+    for arg in i:
+        a = arg.atom
+        if a is None:
+            break
+        vars.append(a)
+        # no condition (currently) has more than 3 arguments. Additional
+        # arguments are allowed but ignored
+        if len(vars) > 3:
```

Comment:
Should this actually be `>= 3` if we only support 3 args? Currently, this ends up with a max of 4 I believe